-1 
Tested at: 6ce37ff79f2e11713639beea766525fbdac4d41b
 I found an error when building: 
 
 <pre>>> Building edm plugin tmp/slc6_amd64_gcc493/src/RecoLocalMuon/RPCRecHit/test/RPCRecHitReader/libRPCRecHitReader.so
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-07-2300/src/RecoLocalMuon/CSCValidation/src/CSCValHists.cc 
Leaving library rule at src/RecoLocalMuon/RPCRecHit/test
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-07-2300/src/RecoLocalMuon/CSCValidation/src/CSCValidation.cc 
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-07-2300/src/HeavyFlavorAnalysis/Onia2MuMu/src/OniaVtxReProducer.cc: In constructor 'OniaVtxReProducer::OniaVtxReProducer(const edm::Handle<std::vector<reco::Vertex> >&, const edm::Event&)':
/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-07-2300/src/HeavyFlavorAnalysis/Onia2MuMu/src/OniaVtxReProducer.cc:29:48: error: 'const class edm::Provenance' has no member named 'parents'
     std::vector<edm::BranchID> parents = prov->parents();
                                                ^
gmake: *** [tmp/slc6_amd64_gcc493/src/HeavyFlavorAnalysis/Onia2MuMu/src/HeavyFlavorAnalysisOnia2MuMu/OniaVtxReProducer.o] Error 1
>> Compiling edm plugin /tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-07-2300/src/RecoLocalMuon/GEMSegment/test/TestME0SegmentAnalyzer.cc 
>> Building edm plugin tmp/slc6_amd64_gcc493/src/RecoLocalMuon/GEMSegment/plugins/RecoLocalMuonGEMSegmentAuto/libRecoLocalMuonGEMSegmentAuto.so
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12894/10446/summary.html

--- a/HeavyFlavorAnalysis/Onia2MuMu/src/OniaVtxReProducer.cc
+++ b/HeavyFlavorAnalysis/Onia2MuMu/src/OniaVtxReProducer.cc
@@ -5,6 +5,7 @@
 #include "TrackingTools/Records/interface/TransientTrackRecord.h"
 #include "FWCore/ParameterSet/interface/Registry.h"
 #include "FWCore/Common/interface/Provenance.h"
+#include "DataFormats/Provenance/interface/ProductProvenance.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 OniaVtxReProducer::OniaVtxReProducer(const edm::Handle<reco::VertexCollection> &handle, const edm::Event &iEvent) 
@@ -26,7 +27,7 @@ OniaVtxReProducer::OniaVtxReProducer(const edm::Handle<reco::VertexCollection> &
     configure(psetFromProvenance); 
 
     // Now we also dig out the ProcessName used for the reco::Tracks and reco::Vertices
-    std::vector<edm::BranchID> parents = prov->parents();
+    std::vector<edm::BranchID> parents = prov->productProvenance()->parentage().parents();
     bool foundTracks = false;
     bool foundBeamSpot = false;
     for (std::vector<edm::BranchID>::const_iterator it = parents.begin(), ed = parents.end(); it != ed; ++it) {

